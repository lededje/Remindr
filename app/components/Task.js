import React from 'react';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';

export default function Task(props) {
  return (
    <View style={styles.item} testID="Task">
      <Text style={styles.title}>{props.title}</Text>
      <Text style={styles.timestamp}>{props.timestamp}</Text>
    </View>
  );
}

Task.propTypes = {
  id: React.PropTypes.number.isRequired,
  title: React.PropTypes.string.isRequired,
  timestamp: React.PropTypes.number.isRequired,
};

const styles = StyleSheet.create({
  item: {
    paddingLeft: 35,
    paddingRight: 35,
    paddingTop: 20,
    paddingBottom: 20,
    backgroundColor: '#fff',
    borderColor: '#eee',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  title: {
    fontSize: 14,
    marginBottom: 10,
    fontWeight: '300',
  },
  timestamp: {
    fontSize: 12,
    color: '#999',
    fontWeight: '300',
  },
});
