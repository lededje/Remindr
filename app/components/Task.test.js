import React from 'react';
import Task from './Task';
import renderer from 'react-test-renderer';

it('renders the task component', () => {
  const component = renderer.create(
    <Task id={123} title="Test Task" timestamp={1474160934141} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
