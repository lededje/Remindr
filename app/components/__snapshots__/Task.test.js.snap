exports[`test renders the task component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#eee",
      "paddingBottom": 20,
      "paddingLeft": 35,
      "paddingRight": 35,
      "paddingTop": 20
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 14,
        "fontWeight": "300",
        "marginBottom": 10
      }
    }>
    Test Task
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#999",
        "fontSize": 12,
        "fontWeight": "300"
      }
    }>
    1474160934141
  </Text>
</View>
`;
